src/main/java/com/dragonballs/services/user/UserService.java

```diff
@@ -0,0 +1,49 @@
+package com.dragonballs.services.user;
+
+import com.dragonballs.dao.UserDAO;
+import com.dragonballs.entities.User;
+import com.dragonballs.exceptions.UserException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class UserService {
```

Comment:
It should be added in multiple places, but in there it might be critical - Need logging
https://www.baeldung.com/spring-boot-logging